Source/Scene/PostProcessCollection.js

```diff
@@ -0,0 +1,596 @@
+        Check.typeOf.number.greaterThanOrEquals('index', index, 0);
+        Check.typeOf.number.lessThan('index', index, length);
+        //>>includeEnd('debug');
+        return this._processes[index];
+    };
+
+    /**
+     * Removes all processes from the collection and destroys them.
+     */
+    PostProcessCollection.prototype.removeAll = function() {
+        var processes = this._processes;
+        var length = processes.length;
+        for (var i = 0; i < length; ++i) {
+            this.remove(processes[i]);
+        }
+        processes.length = 0;
+    };
+
+    /**
+     * Gets a post-process in the collection by its name.
+     *
+     * @param {String} name The name of the post-process.
+     * @return {PostProcess|PostProcessComposite} The post-process.
+     *
+     * @private
+     */
+    PostProcessCollection.prototype.getProcessByName = function(name) {
+        return this._processNames[name];
+    };
+
+    /**
+     * Gets the framebuffer to be used for a post-process with the given name.
+     *
+     * @param {String} name The name of a post-process.
+     * @return {Framebuffer} The framebuffer to be used for a post-process with the given name.
+     *
+     * @private
+     */
+    PostProcessCollection.prototype.getFramebuffer = function(name) {
+        return this._textureCache.getFramebuffer(name);
+    };
+
+    /**
+     * Called before the processes in the collection are executed. Calls update for each process and creates WebGL resources.
+     *
+     * @param {Context} context The context.
+     *
+     * @private
+     */
+    PostProcessCollection.prototype.update = function(context) {
+        removeProcesses(this);
+
+        var activeProcesses = this._activeProcesses;
+        var processes = this._processes;
+        var length = activeProcesses.length = processes.length;
+
+        var i;
+        var process;
+        var count = 0;
+        for (i = 0; i < length; ++i) {
+            process = processes[i];
+            if (process.ready && process.enabled) {
+                activeProcesses[count++] = process;
+            }
+        }
+        activeProcesses.length = count;
+
+        var ao = this._ao;
+        var bloom = this._bloom;
+        var fxaa = this._fxaa;
+
+        if (this._cacheDirty || count !== this._lastLength || ao.enabled !== this._aoEnabled || bloom.enabled !== this._bloomEnabled || fxaa.enabled !== this._fxaaEnabled) {
+            // The number of processes to execute has changed.
+            // Update dependencies and recreate framebuffers.
+            this._textureCache = this._textureCache && this._textureCache.destroy();
+            this._textureCache = new PostProcessTextureCache(this);
+
+            this._lastLength = count;
+            this._aoEnabled = ao.enabled;
+            this._bloomEnabled = bloom.enabled;
+            this._fxaaEnabled = fxaa.enabled;
+            this._cacheDirty = false;
```

Comment:
Rename to `_textureCacheDirty`?  At first glance, I did not know what the class would be caching.